akka-stream/src/main/scala/akka/stream/impl/SplitAfterProcessorImpl.scala

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright (C) 2009-2014 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.impl
+
+import scala.util.control.NonFatal
+import akka.actor.Props
+import akka.stream.ActorFlowMaterializerSettings
+import akka.stream.Supervision
+import akka.stream.scaladsl.Source
+
+/**
+ * INTERNAL API
+ */
+private[akka] object SplitAfterProcessorImpl {
+  def props(settings: ActorFlowMaterializerSettings, splitPredicate: Any ⇒ Boolean): Props =
+    Props(new SplitAfterProcessorImpl(settings, splitPredicate))
+
+  private trait SplitDecision
```

Comment:
So it's here on [wip-one-impl-to-rule-them-all-ktoso](https://github.com/ktoso/akka/commit/9fabf3cc35471a91642473a1aac50f9e47953850).
Though I think it's gotten more complicated (2 separate modes, mixted together) and more allocations. 

Could be cleaned up a bit, but first wanted feedback if we prefer one impl or two simple ones. Please chime in guys @drewhk @rkuhn @patriknw (2 simple impls is this PR, 1 smart impl is the above commit).